src/main/java/com/google/devtools/build/lib/bazel/rules/cpp/BazelCppSemantics.java

```diff
@@ -78,15 +78,15 @@ public void finalizeCompileActionBuilder(
       RuleErrorConsumer ruleErrorConsumer) {
     CcToolchainProvider toolchain = actionBuilder.getToolchain();
     if (language == Language.CPP) {
+      CppConfiguration cppConfig = configuration.getFragment(CppConfiguration.class);
       actionBuilder
           .addTransitiveMandatoryInputs(
-              configuration.getFragment(CppConfiguration.class).useSpecificToolFiles()
-                      && !actionBuilder.getSourceFile().isTreeArtifact()
+              cppConfig.useSpecificToolFiles() && !actionBuilder.getSourceFile().isTreeArtifact()
                   ? (actionBuilder.getActionName().equals(CppActionNames.ASSEMBLE)
                       ? toolchain.getAsFiles()
                       : toolchain.getCompilerFiles())
                   : toolchain.getAllFiles())
-          .setShouldScanIncludes(false);
+          .setShouldScanIncludes(cppConfig.shouldScanIncludes());
```

Comment:
This brings up an idea. Do you think it would be better to remove the `--experimental_include_scanning` flag and instead have a `experimental_unsupported_include_scanning` feature that users can enable with `--features`? That would be more configurable and more consistent with the internal world. (Or is the condition before the first `&&` you pasted also checking an option?)

Reply:
Yes, the condition before && is also a flag. So I think checking the cc_include_scanning feature and the experimental flag is closer to what we have internally. Even though there aren't CPP_MODULES in Bazel, if you add that to the condition as well then it's a perfect match except for the flag names.